File: PP/Lista1/count.scala
object count {
    def main(args: Array[String]) : Unit = {
        def count[A](x : A, xs : List[A]) : Int =
            (x, xs) match {
                case (_, List()) => 0
                case (x, (h :: t)) => if h == x then 1 + count(x, t) else count(x, t)
            }

        println(count[Char]('a', List('a', 'l', 'a')))
    }
}

File: PP/Lista1/count.ml
(*
    O(n) + O(n) = O(n) 
    a' * a' list -> int
*)
let rec count (x,y) =
    match (x,y) with
    | (_, []) -> 0
    | (x, h :: t) -> if h = x then 1 + count (x,t) else count (x,t);;

print_int (count ('a', ['a';'l';'a']));;
